userspace/libsinsp/sinsp_ppm_sc.cpp

```diff
@@ -166,52 +166,73 @@ std::unordered_set<uint32_t> sinsp::enforce_simple_ppm_sc_set(std::unordered_set
 
 std::unordered_set<uint32_t> sinsp::enforce_io_ppm_sc_set(std::unordered_set<uint32_t> ppm_sc_set)
 {
-	for(int i = 0; i < PPM_SC_MAX; i++)
+	const int bitmask = EC_SYSCALL - 1;
+	for(int ppm_sc_code = 0; ppm_sc_code < PPM_SC_MAX; ppm_sc_code++)
 	{
-		if(g_infotables.m_syscall_info_table[i].category == EC_IO_READ ||
-		   g_infotables.m_syscall_info_table[i].category == EC_IO_WRITE ||
-		   g_infotables.m_syscall_info_table[i].category == EC_IO_OTHER ||
-		   g_infotables.m_syscall_info_table[i].category == EC_FILE)
+		switch(g_infotables.m_syscall_info_table[ppm_sc_code].category & bitmask)
```

Comment:
I think we could just use `& EC_IO_BASE` right? Like
```
for(int i = 0; i < PPM_SC_MAX; i++)
	{
		if(g_infotables.m_syscall_info_table[i].category & EC_IO_BASE)
```

Reply:
But i am ok with the changes, since we are not perf driven here, therefore it is ok to follow the same code logic of the other helpers.

Reply:
First tried this type of approach and it wasn't working consistently, it tripped me off in the other PR. Then stumbled across https://github.com/falcosecurity/libs/blob/38cd13866be524e1c50058bc44d38598217dfd09/userspace/libsinsp/event.h#L253 and this approach always returns correct results.

Reply:
Very weird about `EC_IO_BASE`; which issues have you seen?